src/serial.rs

```diff
@@ -812,9 +846,18 @@ mod tests {
     #[test]
     fn test_serial_events() {
         let intr_evt = EventFd::new(libc::EFD_NONBLOCK).unwrap();
-        let metrics = Arc::new(ExampleSerialMetrics::default());
+
+        let events_ = Arc::new(ExampleSerialEvents::new());
         let mut oneslot_buf = [0u8; 1];
-        let mut serial = Serial::with_events(intr_evt, metrics, oneslot_buf.as_mut());
+        let mut serial = Serial::with_events(intr_evt, events_, oneslot_buf.as_mut());
+
+        assert!(serial
+            .events
+            .buffer_ready_event
+            .as_ref()
+            .unwrap()
+            .read()
+            .is_err());
```

Comment:
Let's check for the actual error instead of just that a generic error is returned. In this case, because io::Error does not implement `PartialEq`, we can use `.kind` as follows:
```
        assert_eq!(
            serial
                .events
                .buffer_ready_event
                .as_ref()
                .unwrap()
                .read()
                .unwrap_err()
                .kind(),
            ErrorKind::WouldBlock
        );
```